python/detectify-2.7.py

```diff
@@ -93,34 +95,104 @@ def get_domains(api_key, secret_key):
     path = "/domains/"
     req = request(api_key, secret_key, path, 'GET')
     if req != None:
-        print eval(req.text)
+        print json.loads(req.text)
 
 def get_domain_profiles(api_key, secret_key, domain_token):
     path = "/profiles/"+domain_token+"/"
     req = request(api_key, secret_key, path, 'GET')
     if req != None:
-        print eval(req.text)
+        print json.loads(req.text)
 
 def get_all_profiles(api_key, secret_key):
     path = "/profiles/"
     req = request(api_key, secret_key, path, 'GET')
     if req != None:
-        return eval(req.text)
+        return json.loads(req.text)
 
 def profiles_scan_status(api_key, secret_key):
-    profiles = get_all_profiles(api_key, secret_key);
+    profiles = get_all_profiles(api_key, secret_key)
     if profiles != None:
         for scan in profiles:
             #print i["token"]
-            scan_status(scan["token"], api_key, "")
+            scan_status(scan["token"], api_key, secret_key)
 
+def get_scan_reports(api_key, secret_key, profile):
+    path = "/reports/"+profile+"/"
+    req = request(api_key, secret_key, path, 'GET')
+    if req != None:
+        return json.loads(req.text)
+
+def get_findings_report(api_key, secret_key, profile, report):
+    path = "/fullreports/"+ profile +"/"+report+"/"
+    req = request(api_key, secret_key, path, 'GET')
+    if req != None:
+        return json.loads(req.text)
+
+def profiles_findings(api_key, secret_key):
+    #We get all scan profiles
+    profiles = get_all_profiles(api_key, secret_key)
+    reports = []
+    all_reports = {}
+
+    #We get all the reports of all profiles, and
+    #construct a dict with that info
+    for profile in profiles:
+        #reports on profile
+        reports = get_scan_reports(api_key, secret_key, profile["token"])
+        all_reports[profile["endpoint"]] = {"token": profile["token"], "reports": list([x["token"] for x in reports])}
+    
+    total = 0
+    for i in all_reports.keys():
+        total += len(all_reports[i]["reports"])
+    
+    #Gets the files already downloaded to skip them
+    report_files = []
+    for (dirpath, dirnames, filenames) in walk("."):
+        report_files.extend(filenames)
+        report_files = [files for files in report_files if ".json" in files]
+        break
+    
+    #Deletes the reports already downloaded from the download list
+    for files in report_files:
+        scan, report = (files.split(".json")[0]).split("_")
+        if scan in all_reports.keys():
+            if report in all_reports[scan]["reports"]:
```

Comment:
I would recommend to make sure the report was completed when last downloaded, incomplete reports should be replaced. You can use the `stopped` timestamp to check if the report is complete.

Reply:
I was checking now on this, and when downloading the reports, I was expecting that they were already finished before being stored as report; in the case of using the `stopped` timestamp, that value is represented as empty?

Also, at that point of the code I am working with only the filenames, not the full report data, so this check should be done at #L141 or #L142 when I am going through the reports. Will check on that.